kedro/extras/datasets/spark/spark_hive_dataset.py

```diff
@@ -209,8 +226,16 @@ def _save(self, data: DataFrame) -> None:
     def _insert_save(self, data: DataFrame) -> None:
         data.createOrReplaceTempView("tmp")
         columns = ", ".join(self._table_columns)
+
+        partition = ""
+        if self._partition is not None:
+            partition = f"partition ({self._partition.strip()})"
+
```

Comment:
I would probably put this logic in a slightly different place.

e.g. If `partition` argument remains a string then make its default argument `""` and do this in `__init__` method to begin with:
```
self._partition = partition.strip()
```
And then here you'd do:
```
        self._get_spark().sql(
            f"""
            insert into {self._database}.{self._table} partition ({self._partition})
            select {columns} from tmp
            """  # nosec
        )
```

Reply:
I like these. But, when the table is not partitioned, `partition ({self._partition})` isn't gonna work. Maybe we can move that to the `__init__` too?

Reply:
Ah, in that case I would keep `self._partition` as an empty string but still keep the logic for building the SQL query next to the SQL query:
```
        partition =  f"partition ({self._partition})" if self._partition else ""
        self._get_spark().sql(
            f"""
            insert into {self._database}.{self._table} {partition}
            select {columns} from tmp
            """  # nosec
        )
```